File: exercism/go/go/darts/darts.go
// Package calculates the points earned in a darts game
package darts

import "math"

const (
	OUTER_RADIUS  = 10
	MIDDLE_RADIUS = 5
	INNER_RADIUS  = 1
)

// Score calculates score base on the cartesian co-ordinates
func Score(x, y float64) int {
	// inner circle radius of 1 - 10 points
	// middle circle radius of 5 - 5 points
	// outer circle radius of 10 - 1 point
	// outside - 0 points
	dist := math.Sqrt(x*x + y*y)

	if dist > OUTER_RADIUS {
		return 0
	} else if dist > MIDDLE_RADIUS {
		return 1
	} else if dist > INNER_RADIUS {
		return 5
	}

	return 10
}

File: exercism/c/c/darts/src/darts.c
#include "darts.h"
#include <math.h>
#include <stdio.h>

// outer circle - 10 radius 
// middle circle - 5 radius
// inner circle - 1 radius
uint8_t score(coordinate_t points) {

	float res = sqrt(points.x * points.x + points.y * points.y);

	if (res <= 1) {
		return 10;
	} else if (res <= 5) {
		return 5; 
	} else if (res <= 10) {
		return 1;
	} else {
		return 0;
	}
}
